devices.js

```diff
@@ -8661,6 +8661,27 @@ const devices = [
             await configureReporting.currentSummReceived(endpoint);
         },
     },
+
+    // Smartenit
+    {
+        zigbeeModel: ['ZBMLC30'],
+        model: '4040B',
+        vendor: 'Smartenit',
+        description: 'Wireless Metering 30A Dual-Load Switch / Controller',
+        supports: 'on/off',
+        fromZigbee: [fz.on_off, fz.ZBMLC30_power],
+        toZigbee: [tz.on_off],
+        meta: {configureKey: 1},
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 2};
+        },
+        configure: async (device, coordinatorEndpoint) => {
+            const endpoint1 = device.getEndpoint(1);
+            await bind(endpoint1, coordinatorEndpoint, ['genOnOff', 'seMetering']);
+            const endpoint2 = device.getEndpoint(2);
+            await bind(endpoint2, coordinatorEndpoint, ['genOnOff']);
```

Comment:
Can you add `await readMeteringPowerConverterAttributes(endpoint);` here and replace `ZBMLC30_power` with `metering_power`. Does this work?

Reply:
I get this error:
```
Mar 26 17:30:44 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:30:44: Configuring 'shop_bench_240'
Mar 26 17:30:59 automate npm[9297]: zigbee2mqtt:error 2020-03-26 17:30:59: Failed to configure 'shop_bench_240', attempt 1 (Error: Read 0x000b52000000497f/1 seMetering(["multiplier","divisor"], {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":true}) failed (Error: Timeout - 5591 - 1 - 6 - 1794 - 1 after 6000ms)
Mar 26 17:30:59 automate npm[9297]:     at Endpoint.<anonymous> (/root/zigbee-herdsman/dist/controller/model/endpoint.js:173:23)
Mar 26 17:30:59 automate npm[9297]:     at Generator.throw (<anonymous>)
Mar 26 17:30:59 automate npm[9297]:     at rejected (/root/zigbee-herdsman/dist/controller/model/endpoint.js:6:65))
```

Reply:
Can you try to repair the device?

Reply:
```
Mar 26 17:37:19 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:19: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"device_connected","message":{"friendly_name":"0x000b52000000497f"}}'
Mar 26 17:37:19 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:19: Starting interview of '0x000b52000000497f'
Mar 26 17:37:19 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:19: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"interview_started","meta":{"friendly_name":"0x000b52000000497f"}}'
Mar 26 17:37:19 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:19: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"device_announced","message":"announce","meta":{"friendly_name":"0x000b52000000497f"}}'
Mar 26 17:37:22 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:22: Successfully setup reporting for '0x000b52000000497f' - 1 - genOnOff
Mar 26 17:37:22 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:22: Successfully setup reporting for '0x000b52000000497f' - 1 - genLevelCtrl
Mar 26 17:37:33 automate npm[9297]: zigbee2mqtt:error 2020-03-26 17:37:33: Failed to interview '0x000b52000000497f', device has not successfully been paired
Mar 26 17:37:33 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:33: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"interview_failed","meta":{"friendly_name":"0x000b52000000497f"}}'
Mar 26 17:37:33 automate npm[9297]: zigbee2mqtt:info  2020-03-26 17:37:33: Configuring '0x000b52000000497f'
Mar 26 17:37:50 automate npm[9297]: zigbee2mqtt:error 2020-03-26 17:37:50: Failed to configure '0x000b52000000497f', attempt 2 (Error: Read 0x000b52000000497f/1 seMetering(["multiplier","divisor"], {"timeout":6000,"manufacturerCode":null,"disableDefaultResponse":true}) failed (Error: Timeout - 43471 - 1 - 47 - 1794 - 1 after 6000ms)
Mar 26 17:37:50 automate npm[9297]:     at Endpoint.<anonymous> (/root/zigbee-herdsman/dist/controller/model/endpoint.js:173:23)
Mar 26 17:37:50 automate npm[9297]:     at Generator.throw (<anonymous>)
Mar 26 17:37:50 automate npm[9297]:     at rejected (/root/zigbee-herdsman/dist/controller/model/endpoint.js:6:65))
```

Still can't read those attributes.

Hmm, why does it set up reporting for `genLevelCtrl`?

Reply:
![image](https://user-images.githubusercontent.com/5122115/77755528-d3de5e80-7003-11ea-9a3f-bda780214934.png)

It responds to the bind request, but it simply ignores the attribute read.

Reply:
It doesn't respond indeed.

Please try with the following definition:

```js
// Smartenit
{
    zigbeeModel: ['ZBMLC30'],
    model: '4040B',
    vendor: 'Smartenit',
    description: 'Wireless Metering 30A Dual-Load Switch / Controller',
    supports: 'on/off, power measurements',
    fromZigbee: [fz.on_off, fz.metering_power],
    toZigbee: [tz.on_off],
    meta: {configureKey: 2},
    endpoint: (device) => {
        return {l1: 1, l2: 2};
    },
    configure: async (device, coordinatorEndpoint) => {
        const endpoint1 = device.getEndpoint(1);
        await bind(endpoint1, coordinatorEndpoint, ['genOnOff', 'seMetering']);
        const endpoint2 = device.getEndpoint(2);
        await bind(endpoint2, coordinatorEndpoint, ['genOnOff']);

        // Device doesn't respond to divisor read, set it here
        // https://github.com/Koenkk/zigbee-herdsman-converters/pull/1096
        endpoint1.saveClusterAttributeKeyValue('seMetering', {
            divisor: 100,
            multiplier: 1,
        });
    },
},
```

Reply:
That worked.  I did need to change the divisor because it gets multiplied again by 1000 in the converter.  Also I needed to bind seMetering on endpoint 2 or else it wouldn't report values when one of the outputs was turned off.  It's very strange.  It only has one current sensor on the input.  Also I told it to ignore the brightness reports because it was sending those, which doesn't make sense for a box with relays in it.  Anyway seems to be working fine with the latest changes.